ratis-logservice/src/main/java/org/apache/ratis/logservice/api/AsyncLogReader.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.logservice.api;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Asynchronous client interface to read from a LogStream.
+ */
+public interface AsyncLogReader extends AutoCloseable {
+
+  /**
+   * Seeks to the position before the record at the provided {@code recordId} in the LogStream.
+   *
+   * @param recordId A non-negative, recordId in the LogStream
+   * @return A future for when the operation is completed.
+   */
+  CompletableFuture<Void> seek(long recordId) throws IOException;
+
```

Comment:
Can add reset() also to reset according to the logStream getFirstRecordId().

Reply:
If you think that would be helpful, we can. I am wondering about the frequency of use of such an API. Does it make sense to have that "wrapper" when we have something like `seek(getFirstRecordId())` already present? I don't feel strongly either way :)